includes/Core/Util/Tracking.php

```diff
@@ -22,79 +20,31 @@
  * @access private
  * @ignore
  */
-final class Tracking {
+final class Tracking extends User_Setting {
```

Comment:
I don't think this class should become a `User_Setting`, as it handles quite a bit more. It has been reduced a lot via #1055, but the scope of this class is still bigger than just representing a setting.

Let's instead introduce a `Tracking_Optin` class that extends `User_Setting`. An instance of this class should then be instantiated in the `Tracking` constructor, similar to how `Authentication` instantiates its related options/user options in its constructor.

Reply:
I opted for the name `Tracking_Consent` instead, which I think is more clear and less awkward than `Optin`.